js/symlink_test.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import { test, testPerm, assert, assertEqual } from "./test_util.ts";
+import * as deno from "deno";
+
+testPerm({ write: true }, function symlinkSyncSuccess() {
+  const testDir = deno.makeTempDirSync() + "/test-symlink-sync";
+  const oldname = testDir + "/oldname";
+  const newname = testDir + "/newname";
+  deno.mkdirSync(oldname);
+  deno.symlinkSync(oldname, newname);
+  const newNameInfoLStat = deno.lstatSync(newname);
+  const newNameInfoStat = deno.statSync(newname);
+  assert(newNameInfoLStat.isSymlink());
+  assert(newNameInfoStat.isDirectory());
+});
+
+testPerm({ write: false }, function symlinkSyncPerm() {
+  let err;
+  try {
+    const oldname = "/oldbaddir";
+    const newname = "/newbaddir";
+    deno.symlinkSync(oldname, newname);
```

Comment:
Maybe this should be done in a temp dir? It could be that the permission error was from trying to write to /

Reply:
@ry since this test is `{write: false}` we cannot create a temp directory in this test. However, we can remove `/` to avoid confusion!